File: coffeetime-parent/RubyGun/src/21_exception.rb
def str2int(n)
  begin
    result = Integer(n)
    if result == 1
      raise Exception.new('something wrong')
    end
  rescue ArgumentError => error
    puts error.backtrace
    0
  rescue Exception
    n = 67
    retry
  else
    puts 'success'
    result
  ensure
    puts 'finally'
    300
  end
end

puts str2int(10) # 10
puts str2int('10') # 10
puts str2int('1a0') # 0
puts str2int('a10') # 0
puts str2int('1') # 67


def promptAndGet(i)
  for j in i..20
    catch :quit do
      throw :quit if j == 3
      puts "j is #{j}"
    end
  end
end

promptAndGet(5)
promptAndGet(3)

File: coffeetime-parent/Python3Gun/src/21_exception.py
def str2int(n):
    try:
        result = int(n)
        if result == 1:
            raise Exception('something wrong')
    except (ArithmeticError, TypeError) as error:
        print(error)
        return 0
    except Exception as error:
        print('exception: ', error)
        return 67
    else:
        print('no exception')
    finally:
        print('finally')
        return 300


print(str2int(10))  # 300
print(str2int('10'))  # 300
print(str2int('1a0'))  # 300
print(str2int('a10'))  # 300
print(str2int('1'))  # 300
